### ISSUE TITLE
`torch.finfo(torch.float8_e4m3fn).max` crashes python runtime

### ISSUE THREAD

**Issue by malfet** [pytorch/pytorch#109737]
### 🐛 Describe the bug

```
% python -c "import torch;print(torch.finfo(torch.float8_e4m3fn).max)" 
libc++abi: terminating due to uncaught exception of type c10::Error: "max" not implemented for 'Float8_e4m3fn'
Exception raised from operator() at /Users/nshulga/git/pytorch/pytorch/torch/csrc/TypeInfo.cpp:129 (most recent call first):
frame #0: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) + 92 (0x103029670 in libc10.dylib)
frame #1: THPFInfo_max(THPFInfo*, void*) + 228 (0x10494c7a8 in libtorch_python.dylib)
frame #2: _PyObject_GenericGetAttrWithDict + 208 (0x10250e120 in python3.9)
frame #3: _PyEval_EvalFrameDefault + 19188 (0x1025a4014 in python3.9)
frame #4: _PyEval_EvalCode + 2804 (0x10259eec8 in python3.9)
frame #5: run_mod + 216 (0x1025f8c0c in python3.9)
frame #6: PyRun_StringFlags + 172 (0x1025f9c14 in python3.9)
frame #7: PyRun_SimpleStringFlags + 288 (0x1025f9b18 in python3.9)
frame #8: Py_RunMain + 1548 (0x102618668 in python3.9)
frame #9: pymain_main + 1252 (0x102619a40 in python3.9)
frame #10: main + 56 (0x10246c7a0 in python3.9)
frame #11: start + 2236 (0x1ab837f28 in dyld)

zsh: abort      python -c "import torch;print(torch.finfo(torch.float8_e4m3fn).max)"
```

### Versions

nightly, 2.1

cc @yanbing-j @vkuzo @albanD @kadeng

**Comment by albanD** [comment]
Adding the `HANDLE_TH_ERROR` macros to THPFInfo_max will do the trick.

**Comment by malfet** [comment]
@albanD yes, it requires a stack of 2 PR: 
- avoid crash
- extend max/min/etc to float8 dtypes

**Comment by atalman** [comment]
Fix was cherry-picked to 2.1 release https://github.com/pytorch/pytorch/pull/109808 - removing from 2.1.1 milestones
